### ISSUE TITLE
[Docs]: ImageBuilder removes Systems Manager Agent by default

### ISSUE THREAD

**Issue by mocknen** [hashicorp/terraform-provider-aws#31619]
### Documentation Link

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/imagebuilder_image_recipe

### Description

Although this is probably not a problem with the provider itself, I think it is worth documenting thus I am creating this issue.

Regarding the `aws_imagebuilder_image_recipe` resource [documentation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/imagebuilder_image_recipe#uninstall_after_build), it states that `uninstall_after_build` in the `systems_manager_agent` block is `false` by default. And this statement is indeed consistent with the AWS API [documentation](https://docs.aws.amazon.com/imagebuilder/latest/APIReference/API_SystemsManagerAgent.html), as quoted below.

> Controls whether the Systems Manager agent is removed from your final build image, prior to creating the new AMI. If this is set to true, then the agent is removed from the final image. If it's set to false, then the agent is left in, so that it is included in the new AMI. The default value is false.

However, the actual behavior is different: Systems Manager agents are removed by default unless the value is explicitly specified as `false`. I confirmed this behavior by examining the contents of the API request in event history in CloudTrail. If `additionalInstanceConfiguration` block is not included in the request parameters, the instance configuration field does not appear in the image recipe summary in web console, but only when the recipe is created with `uninstallAfterBuild` set to `false`, the notation shown in the image below appears.

![20230529124712](https://github.com/hashicorp/terraform-provider-aws/assets/14985501/67b5acb2-a7cd-46b3-bf19-5b68685cc92d)

In summary, differences can be seen between the current provider documentation and the actual behavior, but it is possible that the AWS documentation is wrong in the first place. I think it might be helpful to add a note to the document about this behavior to avoid confusion.

### References

* Action 
  * [CreateImageRecipe](https://docs.aws.amazon.com/imagebuilder/latest/APIReference/API_CreateImageRecipe.html)
* Data type 
  * [AdditionalInstanceConfiguration](https://docs.aws.amazon.com/imagebuilder/latest/APIReference/API_AdditionalInstanceConfiguration.html)
  * [SystemsManagerAgent](https://docs.aws.amazon.com/imagebuilder/latest/APIReference/API_SystemsManagerAgent.html)

### Would you like to implement a fix?

None

**Comment by justinretzolk** [comment]
Hey @mocknen 👋 Thank you for taking the time to raise this! My read on this is that if `uninstall_after_build` is not passed, the Systems Manager agent should not be removed, so this feels like a potential bug. I've marked this as such so that we can take a look into it when this can be prioritized.

**Comment by mokeeffe-digistorm** [comment]
I just spent the best part of a day trying to figure out why our new AMIs were uncontactable by Systems Manager after booting. I would agree this is definitely a bug, either in the Terraform Provider or the AWS API itself. I'm in touch with AWS support to find out if there is any issue on their end.

**Comment by mokeeffe-digistorm** [comment]
AWS support have explained the behaviour on their end:

_tl; dr:_

> If the base instance/AMI doesn't come with the agent, it will be removed before creating images 

There is a note in the documentation https://docs.aws.amazon.com/imagebuilder/latest/userguide/create-image-recipes.html under "additionalInstanceConfiguration" that explains the behaviour:

<img width="977" alt="Image" src="https://github.com/user-attachments/assets/e7d267c1-2c30-4c71-83f3-dd161f445333" />
